tests/features/joinup_discussion/discussion.subscribe.feature

```diff
@@ -75,7 +75,6 @@ Feature: Subscribing to discussions
     And I fill in "Email" with "trollingismylife@example.com"
     But I wait for the honeypot validation to pass
     Then I press "Post comment"
-    Then 0 e-mails should have been sent
```

Comment:
After https://github.com/ec-europa/joinup-dev/pull/1140/commits/71c8c782eee70b2c9b9317ea087d9cc70feb2036 a bug was revealed due to not clearing the state cache.

This step failed as there are test moderators in the site from the beginning and they are receiving the email to validate the comment of the anonymous user.